programs/server/Server.cpp

```diff
@@ -2191,6 +2192,8 @@ try
     if (dns_cache_updater)
         dns_cache_updater->start();
 
+    auto replicas_reconnector = ReplicasReconnector::init(global_context);
```

Comment:
Passing the whole `global_context` looks a bit scary. Maybe we can create a task in BackgroundSchedulePool and pass it to the ReplicasReconnector along with a server setting?